### ISSUE TITLE
"Failed to extract information from neither embed nor watch pages"

### ISSUE THREAD

**Issue by kirk-marple** [dotnettools/SharpGrabber#97]
Anyone know if anything changed recently, and now YouTube videos are getting blocked by this library.

We're getting a GrabParseException with this error "Failed to extract information from neither embed nor watch pages".

```
                var result = await grabber.GrabAsync(uri);
```

This code used to work, and I recently noticed our tests failing, and it seems to be on any YouTube video.

**Comment by javidsho** [comment]
YouTube has made some big changes. I am a bit busy at the moment to handle the issue myself. I hope someone else would try fixing it. One way or another worst case scenario other open source repositories will probably push the fixes and we can replicate those

**Comment by kirk-marple** [comment]
Ah, thought maybe they were totally blocking things.

I'm happy to give it a go and try and fix it.  Any advice on what they have changed would be helpful, and then I can make a PR.

**Comment by dimanrus** [comment]
Hello. If you call GrabAsync again, for example in catch, then everything works

**Comment by rochapablo** [comment]
+1

**Comment by PatriceBG** [comment]
I tried to repeat the GrabAsync for up to 10 times with the same result. Is there anything else I could try?

**Comment by PrinceCharles23** [comment]
am i wrong to assume the library is useless for youtube for over a year now?

**Comment by PatriceBG** [comment]
No, you're right. Personally, I gave up on SharpGrabber for YouTube. However, YouTubeDownloader (https://github.com/Tyrrrz/YoutubeDownloader) worked very well for me. Give it a try.

**Comment by PrinceCharles23** [comment]
> No, you're right. Personally, I gave up on SharpGrabber for YouTube. However, YouTubeDownloader (https://github.com/Tyrrrz/YoutubeDownloader) worked very well for me. Give it a try.

i considered them until i saw that they want to shove their political agenda down my throat and covering it as "usage license". i am using [YoutubeDLSharp](https://github.com/Bluegrams/YoutubeDLSharp) which also supports separate audio/video downloads ;-)
